### PR TITLE
feat(tasks/prettier-conformance): parse the specs and pass the correct options to prettier

### PR DESCRIPTION
**Pull request by Dunqing**
### This PR did the following things.

1. Parse specs
2. Collect `run_spec's` arguments
3. Pass the correct `PrettierOptions` to the `Prettier`
4. Printing snapshots correctly

### CODE REVIEW

**Comment by Dunqing** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1327** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/1327?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/oxc-project/oxc/1327?utm_source=stack-comment).

File: tasks/prettier_conformance/src/lib.rs [hunk 1]
```diff
@@ -22,9 +28,27 @@ pub struct TestRunnerOptions {
     pub filter: Option<String>,
 }
 
+#[derive(Default)]
```
Comment (Boshen) [inline tasks/prettier_conformance/src/lib.rs:31]: Shall we move this to another file?

**Review by Boshen — CHANGES_REQUESTED** [review]
This is amazing! Let's clean the code up a little bit before merging.

File: tasks/prettier_conformance/src/lib.rs (hunk 1 above)
Comment (Dunqing) [reply tasks/prettier_conformance/src/lib.rs]: Yes, sure

**Review by Boshen — APPROVED** [review]

**Pull request merged**